nixos/modules/services/torrent/transmission.nix

```diff
@@ -100,56 +115,180 @@ in
+        ProtectHome = mkDefault true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectSystem = mkDefault "strict";
+        ReadWritePaths = [ stateDir ];
+        RemoveIPC = true;
+        RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" "AF_INET6" ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        # In case transmission crashes with status=31/SYS,
+        # having systemd.coredump.enable = true
+        # and environment.enableDebugInfo = true
+        # enables to use coredumpctl debug to find the denied syscall.
+        SystemCallFilter = [
+          "@default"
+          "@aio"
+          "@basic-io"
+          #"@chown"
+          #"@clock"
+          #"@cpu-emulation"
+          #"@debug"
+          "@file-system"
+          "@io-event"
+          #"@ipc"
+          #"@keyring"
+          #"@memlock"
+          #"@module"
+          #"@mount"
+          "@network-io"
+          #"@obsolete"
+          #"@pkey"
+          #"@privileged"
+            # Reached when querying infos through RPC (eg. with stig)
+            "quotactl"
+          "@process"
+          #"@raw-io"
+          #"@reboot"
+          #"@resources"
+          #"@setuid"
+          "@signal"
+          #"@swap"
+          "@sync"
+          "@system-service"
+          "@timer"
+        ];
+        SystemCallArchitectures = "native";
+        UMask = "0077";
+      };
     };
 
     # It's useful to have transmission in path, e.g. for remote control
     environment.systemPackages = [ pkgs.transmission ];
 
     users.users = optionalAttrs (cfg.user == "transmission") ({
       transmission = {
-        name = "transmission";
         group = cfg.group;
         uid = config.ids.uids.transmission;
         description = "Transmission BitTorrent user";
-        home = homeDir;
-        createHome = true;
+        home = stateDir;
+        createHome = false;
       };
     });
 
     users.groups = optionalAttrs (cfg.group == "transmission") ({
       transmission = {
-        name = "transmission";
         gid = config.ids.gids.transmission;
       };
     });
 
-    # AppArmor profile
+    networking.firewall = mkIf cfg.openFirewall (
+      if cfg.settings.peer-port-random-on-start
+      then
+        { allowedTCPPortRanges =
+            [ { from = cfg.settings.peer-port-random-low;
+                to   = cfg.settings.peer-port-random-high;
+              }
+            ];
+          allowedUDPPortRanges =
+            [ { from = cfg.settings.peer-port-random-low;
+                to   = cfg.settings.peer-port-random-high;
+              }
+            ];
+        }
+      else
+        { allowedTCPPorts = [ cfg.settings.peer-port ];
+          allowedUDPPorts = [ cfg.settings.peer-port ];
+        }
+    );
+
+    boot.kernel.sysctl = mkIf cfg.settings.utp-enabled {
```

Comment:
Would be nice to leave a comment where these values are taken from.

Reply:
Right, I've added a comment.